core/src/main/java/io/aiven/klaw/service/BaseOverviewService.java

```diff
@@ -78,9 +84,132 @@ protected List<AclInfo> getAclInfoList(
+    while (aclInfo.size() > 0) {
+      String environmentName = aclInfo.get(0).getEnvironmentName();
+      List<AclInfo> envList = new ArrayList<>();
+      List<AclInfo> unprocessable = new ArrayList<>();
+      for (AclInfo acls : aclInfo) {
+        processAclToCorrectList(
+            acls.getEnvironmentName(), environmentName, envList, acls, unprocessable);
+      }
+      organisedList.addAll(mergeReduceAcls(envList));
+      removeProcessedAclsFromList(aclInfo, envList, unprocessable);
+    }
+    return organisedList;
+  }
+
+  private static void removeProcessedAclsFromList(
+      List<AclInfo> aclInfo, List<AclInfo> envList, List<AclInfo> unprocessable) {
+    aclInfo.removeAll(envList);
+    aclInfo.removeAll(unprocessable);
+  }
+
+  private static void processAclToCorrectList(
+      String acls,
+      String environmentName,
+      List<AclInfo> envList,
+      AclInfo acls1,
+      List<AclInfo> unprocessable) {
+    if (acls.equals(environmentName)) {
+      envList.add(acls1);
+    } else if (acls == null) {
+      unprocessable.add(acls1);
+    }
+  }
+
+  private List<AclInfo> mergeReduceAcls(List<AclInfo> aclInfo) {
+    AclInfo producerIp = null, producerPrincipal = null;
+    AclInfo consumerIp = null, consumerPrincipal = null;
+
+    for (AclInfo info : aclInfo) {
+      if (info.getTopictype().equals(AclType.PRODUCER.value)) {
+        if (info.getAcl_ip() != null && !info.getAcl_ip().isEmpty()) {
+          producerIp = initializeAclInfo(producerIp, info);
+          producerIp.getAcl_ips().add(info.getAcl_ip());
+        } else if (info.getAcl_ssl() != null && !info.getAcl_ssl().isEmpty()) {
+          producerPrincipal = initializeAclInfo(producerPrincipal, info);
+          producerPrincipal.getAcl_ssls().add(info.getAcl_ssl());
+        }
+      } else if (info.getTopictype().equals(AclType.CONSUMER.value)) {
+        if (info.getAcl_ip() != null && !info.getAcl_ip().isEmpty()) {
+          consumerIp = initializeAclInfo(consumerIp, info);
+          consumerIp.getAcl_ips().add(info.getAcl_ip());
+        } else if (info.getAcl_ssl() != null && !info.getAcl_ssl().isEmpty()) {
+          consumerPrincipal = initializeAclInfo(consumerPrincipal, info);
+          consumerPrincipal.getAcl_ssls().add(info.getAcl_ssl());
+        }
+      }
+    }
+
+    return addToListIfNotNull(producerIp, producerPrincipal, consumerIp, consumerPrincipal);
+  }
+
+  private List<AclInfo> addToListIfNotNull(
+      AclInfo producerIp,
+      AclInfo producerPrincipal,
+      AclInfo consumerIp,
+      AclInfo consumerPrincipal) {
+    List<AclInfo> buildList = new ArrayList<>();
+    if (producerIp != null) {
+      buildList.add(producerIp);
+    }
+    if (consumerIp != null) {
+      buildList.add(consumerIp);
+    }
+    if (producerPrincipal != null) {
+      buildList.add(producerPrincipal);
+    }
+    if (consumerPrincipal != null) {
+      buildList.add(consumerPrincipal);
+    }
+    return buildList;
```

Comment:
i think we are missing consumer groups

Reply:
yeah we are missing consumer groups will add them in.

Reply:
Are you looking to merge the consumer-groups into each producer/consumer entry or are you looking to make it like.
Team1
	Producer Acls
		1 record : All IP addresses
		1 record : All Principals
	Consumer Acls
		1 record : All IP addresses
		1 record : All Principals
Team2
	Producer Acls
		1 record : All IP addresses
		1 record : All Principals
	Consumer Acls
                Unique Consumer group
        		1 record : All IP addresses
         		1 record : All Principals